components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/filter/AuthParameterFilter.java

```diff
@@ -0,0 +1,190 @@
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authentication.endpoint.util.filter;
+
+import com.google.gson.Gson;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.identity.application.authentication.endpoint.util.AuthContextAPIClient;
+import org.wso2.carbon.identity.application.authentication.endpoint.util.Constants;
+import org.wso2.carbon.identity.application.authentication.endpoint.util.client.model.AuthenticationRequestWrapper;
+import org.wso2.carbon.identity.core.ServiceURLBuilder;
+import org.wso2.carbon.identity.core.URLBuilderException;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+
+import java.io.IOException;
+import java.util.Map;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * Cache the parameters required in the authentication flow.
+ */
+public class AuthParameterFilter implements Filter {
+
+    private static final Log log = LogFactory.getLog(AuthParameterFilter.class);
+    private ServletContext context = null;
+
+    @Override
+    public void init(FilterConfig filterConfig) {
+        context = filterConfig.getServletContext();
+    }
+
+    /**
+     * @param servletRequest
+     * @param servletResponse
+     * @param filterChain
+     * @throws IOException
+     * @throws ServletException
+     */
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain)
+            throws IOException, ServletException {
+
+        // Check if request contains 'sessionDataKey'.
+        if (servletRequest.getParameter("sessionDataKey") != null) {
+            if (servletRequest instanceof HttpServletRequest) {
+                // Cast ServletRequest to HttpServletRequest and get the wrapped request.
+                HttpServletRequest httpServletRequest = cacheParamsFromSessionDataKey((HttpServletRequest) servletRequest);
+                filterChain.doFilter(httpServletRequest, servletResponse);
+                return;
+            }
+        } else if (servletRequest.getParameter("sessionDataKeyConsent") != null) {
```

Comment:
since we are returning in each if and else-if path, can we write these as three if conditions only. That will make the code much readable